rclcpp/include/rclcpp/node.hpp

```diff
@@ -972,7 +972,16 @@ class Node : public std::enable_shared_from_this<Node>
 
   /// Return a list of parameters with any of the given prefixes, up to the given depth.
   /**
-   * \todo: properly document and test this method.
+   * prefixes are a way to highlight namespaces within parameter naming conventions,
+   * they have a hierarchy which is currently separated by "." (a dot).
+   *
+   * \param[in] prefixes The list of prefixes that should be searched for within the
+   * current parameters. If this vector of prefixes is empty, then list_parameters
+   * will return all parameters.
+   * \param[in] depth An unsigned integer that represents the recursive depth to search.
+   * If this depth = 0, then we will return all parameters that fit the prefixes.
```

Comment:
```suggestion
   * If this depth = 0, then all parameters that fit the prefixes will be returned.
```

(keep it in passive voice)